extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/InfinispanClientProcessor.java

```diff
@@ -109,9 +111,24 @@ InfinispanPropertiesBuildItem setup(ApplicationArchivesBuildItem applicationArch
         systemProperties.produce(new SystemPropertyBuildItem("io.netty.noUnsafe", "true"));
         hotDeployment.produce(new HotDeploymentWatchedFileBuildItem(META_INF + File.separator + HOTROD_CLIENT_PROPERTIES));
 
+        // add per cache file config
+        for (InfinispanClientBuildTimeConfig.RemoteCacheConfig cacheConfig : infinispanClient.cache.values()) {
+            if (cacheConfig.configurationResource.isPresent()) {
+                hotDeployment.produce(new HotDeploymentWatchedFileBuildItem(cacheConfig.configurationResource.get()));
+            }
+        }
+
         // Enable SSL support by default
         sslNativeSupport.produce(new ExtensionSslNativeSupportBuildItem(Feature.INFINISPAN_CLIENT));
         nativeImageSecurityProviders.produce(new NativeImageSecurityProviderBuildItem(SASL_SECURITY_PROVIDER));
+        // adds the resource to the native image
+        resourceBuildItem.produce(new NativeImageResourceBuildItem(META_INF + "/" + HOTROD_CLIENT_PROPERTIES));
```

Comment:
Should be `File.separator` like the hotDeployment code above for client properties?